### PR TITLE
Update CIME to ESMCI cime5.8.2-2

### PR DESCRIPTION
**Pull request by jgfouca**
Update CIME to ESMCI cime5.8.2-2

Squash merge of jgfouca/branch-for-to-acme-2019-05-06

Features
* Nuopc update
* nldas_forcing_v2: This is a regional, 0.125 degree data set over the U.S., for use in forcing CTSM.
* Change DATM_CLMNCEP_YR_ALIGN for present-day compsets:  Make DATM_CLMNCEP_YR_ALIGN match DATM_CLMNCEP_YR_START. This will make the model year align with the forcing year.  For
example, if RUN_STARTDATE is set to 2000-01-01, then the model will use year-2000 atmospheric forcing data in the first year, etc.
non-BFB for any present-day compsets using datm (mostly I-cases and G-cases).

Bug fix:
* Fixes to NAG flags in scripts/Tools/Makefile

[non-BFB] (datm)

### CODE REVIEW

File: cime/scripts/Tools/Makefile [hunk 1]
```diff
@@ -677,7 +648,7 @@ CMAKE_OPTS += -D CMAKE_Fortran_FLAGS:STRING="$(FFLAGS) $(EXTRA_PIO_FPPDEFS) $(IN
               -D CMAKE_VERBOSE_MAKEFILE:BOOL=ON \
               -D GPTL_PATH:STRING=$(INSTALL_SHAREDPATH) \
               -D PIO_ENABLE_TESTS:BOOL=OFF \
-              -D PIO_USE_MALLOC:BOOL=ON \
+              -D PIO_USE_MALLOC:BOOL=OFF \
```
Comment (jgfouca) [inline cime/scripts/Tools/Makefile]: @jayeshkrishna

**Comment by rljacob** [comment]
There is a bunch of changes to X and data models.  Can you identify and add to the description?

**Comment by jgfouca** [comment]
@rljacob , the commit messages indicated that the changes were for CESM, but maybe they impact us too.

**Comment by jgfouca** [comment]
These tests diff'd:

```
DIFF ERS_Ld3.f45_f45.ICLM45ED.melvin_gnu.clm-fates (phase RUN)
DIFF SMS_Ly2_P1x1.1x1_smallvilleIA.ICLM45CNCROP.melvin_gnu.clm-force_netcdf_pio (phase RUN)
DIFF ERS.f09_g16.ICLM45BC.melvin_gnu (phase RUN)
DIFF SMS.f09_g16_a.IGCLM45_MLI.melvin_gnu (phase RUN)
DIFF ERS.f19_f19.ICLM45.melvin_gnu (phase RUN)
```

**Comment by rljacob** [comment]
@jonbob and @thorntonpe.  Would like your ok on above datm changes.

**Comment by jonbob** [comment]
@rljacob - we would need time to see if this impacts our current G-cases

**Comment by jgfouca** [comment]
@jonbob I'm trying to maintain a weekly "tick-tock" of alternating CIME splits and merges. If you think it will take a while to verify this PR, let me know and I'll plan for a 2-week cycle instead.

**Comment by jonbob** [comment]
@jgfouca - I'm leaving on vacation on Thursday and not sure I can get it done before then. So we can either wait or could you run the extra_coverage tests? We have two G-cases in that suite and if those pass the BFB test, I don't think there would be any problem with accepting as-is. If they fail, it may have to wait a week or so until I get back?

**Comment by jgfouca** [comment]
@jonbob , will do, thx.

**Comment by rljacob** [comment]
Do you mean the PEM and ERS tests for the G-cases in extra coverage?  We don't have baseliness for that suite.

**Comment by jgfouca** [comment]
@rljacob , i can make some off of master.

**Comment by jgfouca** [comment]
@jonbob 

```
Test 'ERS_P480_Ld5.T62_oEC60to30v3wLI.GMPAS-DIB-IAF-ISMF.sandiatoss3_intel' finished with status 'PASS'
    Path: /gpfs1/jgfouca/acme_scratch/sandiatoss3/ERS_P480_Ld5.T62_oEC60to30v3wLI.GMPAS-DIB-IAF-ISMF.sandiatoss3_intel.C.20190507_142908_48tua2/TestStatus
Test 'PEM_P480_Ld5.T62_oEC60to30v3wLI.GMPAS-DIB-IAF-ISMF.sandiatoss3_intel' finished with status 'PASS'
    Path: /gpfs1/jgfouca/acme_scratch/sandiatoss3/PEM_P480_Ld5.T62_oEC60to30v3wLI.GMPAS-DIB-IAF-ISMF.sandiatoss3_intel.C.20190507_142908_48tua2/TestStatus
```

**Comment by jonbob** [comment]
@jgfouca - if it has no impact on our current compsets, then I think it's fine. Let me check with one person from the ocean group and I think we can sign off on this. Thanks for taking on the baselines

**Comment by jonbob** [comment]
@jgfouca - I checked with @vanroekel and we both agreed that if this PR doesn't impact any of our current compsets, we're fine. So I'll add my review. Thanks for all the testing help

**Comment by jgfouca** [comment]
Thanks @jonbob

**Comment by jgfouca** [comment]
Squash-merged to next.

**Comment by rljacob** [comment]
@jgfouca were those tests against baselines?  I'm pretty sure this will change answers for G-cases.

**Comment by jonbob** [comment]
@rljacob - I think it only impacts present-day usage of datm, which we don't run right now

**Comment by jgfouca** [comment]
@rljacob , yes. I generated my own set of baselines from master and then compared against those using this branch.

**Comment by rljacob** [comment]
Recording which tests for non-bfb from the change to DATM and the year/datm config.
SMS_Ly2_P1x1.1x1_smallvilleIA.ICLM45CNCROP: 2000_DATM%QIA
SMS.f09_g16_a.IGCLM45_MLI:  2000_DATM%QIA
ERS_Ld3.f45_f45.ICLM45ED:  2000_DATM%QIA
ERS.f19_f19.ICLM45:  2000_DATM%QIA
ERS.f09_g16.ICLM45BC:  2000_DATM%QIA

A bunch of land tests and no diffs.  They either did not use a modern year or a different DATM setting.
1850_DATM%CRU, 20TR_DATM%QIA, 1850_DATM%GSWP3v1

A, G-cases had no diffs because of NYF option.
 SMS.T62_oQU120_ais20.MPAS_LISIO_TEST:  2000_DATM%NYF
 SMS.ne30_f19_g16_rx1.A: 2000_DATM%NYF
PET.f45_g37_rx1.A: 2000_DATM&NYF

**Comment by jonbob** [comment]
Also please note that the G-cases in the extra coverage suite test the IAF option:
ERS_P480_Ld5.T62_oEC60to30v3wLI.GMPAS-DIB-IAF-ISMF: 2000_DATM%IAF
PEM_P480_Ld5.T62_oEC60to30v3wLI.GMPAS-DIB-IAF-ISMF: 2000_DATM%IAF

**Pull request merged**